client/components/category-navigation/index.tsx

```diff
@@ -0,0 +1,121 @@
+import { Button } from '@wordpress/components';
+import { useEffect, useState, useRef } from '@wordpress/element';
+import classnames from 'classnames';
+
+import './style.scss';
+
+type CategoryNavProps = {
+	buttons?: {
+		icon: string;
+		label: string;
+		link: string;
+	}[];
+	children?: React.ReactNode;
+	list: {
+		name: string;
+		label?: string;
+		link: string;
+	}[];
+	selectedCategory: string;
+};
+
+export const CategoryNavigation = ( {
+	buttons,
+	children,
+	list,
+	selectedCategory,
+}: CategoryNavProps ) => {
+	const [ showLeftArrow, setShowLeftArrow ] = useState( false );
+	const [ showRightArrow, setShowRightArrow ] = useState( false );
+	const listRef = useRef< HTMLDivElement | null >( null );
+
+	const checkScrollArrows = () => {
+		const list = listRef.current;
+		if ( ! list ) {
+			return;
+		}
+
+		const { scrollLeft, scrollWidth, clientWidth } = list;
+		setShowLeftArrow( scrollLeft > 0 );
+		setShowRightArrow( Math.ceil( scrollLeft ) < scrollWidth - clientWidth );
+	};
+
+	const scrollTo = ( direction: 'right' | 'left' ) => {
+		if ( ! listRef.current ) {
+			return;
+		}
+
+		let left = listRef.current.clientWidth / 2;
+
+		if ( direction === 'left' ) {
+			left *= -1;
+		}
+
+		listRef.current?.scrollBy( { left, behavior: 'smooth' } );
+	};
```

Comment:
I'd consider moving this outside the `CategoryNavigation` scope and passing `listRef.current` as an argument.

Reply:
Sorry, I didn't get your point and reason.

My thoughts - we are creating a common component that should behave the same way everywhere for consistency. So, this should definitely be here to avoid different animations on different pages.

Reply:
This is what I had in mind 33c1d-pb/#js

This is optional, and I don't think we can realistically talk about performance benefits or anything like that with with my suggested approach, it's more a matter of code organization.

`scrollTo` doesn't _have to_ live in the component scope, it could easily live outside, which would make the component a little bit more compact (and easier to read).

Reply:
Oh, got it. My thoguhts are:
1) If we move `scrollTo`, then we should also move `checkScrollArrows` outside.
2) If we don't have a dependency on the DOM node - I would totally agree, but in our case - we have, so I would keep it inside.

Reply:
`checkScrollArrows` has a dependency on state, so I don't think it makes as much sense to move that outside `CategoryNavigation`.

With `scrollTo`, the function isn't as closely tied to any particular ref/state, and it's simple to pass an `element` argument, which is why I think it might make sense in that case.

But if you prefer to keep it as is, then that's fine by me!